src/services/pact-verifier.service.ts

```diff
@@ -12,19 +11,12 @@ export class PactVerifierService {
   ) {}
 
   public async verify(app: INestApplication): Promise<any> {
-    const host = this.options.providerHost || 'localhost';
-    await app.listen(await getPort(), host);
+    const providerUrl = new URL(this.options.providerBaseUrl);
 
-    const appUrl = await app.getUrl();
+    await app.listen(providerUrl.port || 80, providerUrl.hostname);
```

Comment:
This default to `80` is potentially a bug - if the `port` is an empty string, then the port is the default for the protocol, which might not be http.

Also, unless I'm mistaken, this means that you'd need to run tests as root on most systems? This is a breaking change (and I don't think I like it).

If you do remove `getPort`, you can remove the dependency too.